### PR TITLE
feat(internal): global init callback

### PR DESCRIPTION
**Pull request by rarkins**
Adds global `initialize` function (to be followed by `finalize` function) designed so that modules can register their own initialization callbacks to the global worker.

### CODE REVIEW

**Comment by rarkins** [comment]
@viceice @JamieMagee do you think this is the right way to add callbacks? If so then I'll proceed to add the concept of initialize and finalize to both global and repository workers.

File: lib/workers/global/index.ts [hunk 1]
```diff
@@ -64,9 +65,7 @@ export async function start(): Promise<0 | 1> {
     config = await setDirectories(config);
     globalCache.init(config);
     config = await autodiscoverRepositories(config);
-
-    limits.init(config);
```
Comment (rarkins) [inline lib/workers/global/index.ts:68]: Note to self: Forgot to add this to the callbacks

File: lib/workers/global/initialize.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { RenovateConfig } from '../../config/common';
+
+const initializations = [];
```
Comment (rarkins) [inline lib/workers/global/initialize.ts:3]: TODO: add types

**Review by viceice — COMMENTED** [review]
How you make sure the add function is called before init?

You would need to explicitly import each file before.  🤔

**Comment by rarkins** [comment]
> How you make sure the add function is called before init?
> 
> You would need to explicitly import each file before. 🤔

Which files *wouldn't* be imported before the init? Are you thinking about our modules like datasources and versioning? Otherwise I think every file would be imported as part of Node loading it.

**Comment by viceice** [comment]
No, node will load and execute them on demand in order they depend on each other.

**Pull request closed without merging**